From: Ian Beckwith <ianb@erislabs.net>
Date: Sat, 5 Sep 2009 06:48:56 +0100
Subject: Debianize gnulib --version

Last-Update: 2019-10-08

Fix gnulib-tool --version;
also show snapshot and debian package version.
---
 gnulib-tool | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/gnulib-tool b/gnulib-tool
index 2ba5dec..939366b 100755
--- a/gnulib-tool
+++ b/gnulib-tool
@@ -362,12 +362,13 @@ q
     version=' '`cd "$gnulib_dir" && ./build-aux/git-version-gen /dev/null | sed -e 's/-dirty/-modified/'`
   else
     # gnulib copy without versioning information.
-    date=`sed -e 's/ .*//;q' "$gnulib_dir"/ChangeLog`
-    version=
+    date=`zcat /usr/share/doc/gnulib/changelog.gz | sed -e 's/ .*//;q'`
+    version=`zcat /usr/share/doc/gnulib/NEWS.stable.gz | grep -o -m 1 ' [0-9]\+-stable'`
+    debversion=`dpkg-query --show --showformat '${Version}' gnulib`
   fi
   year=`"$gnulib_dir"/build-aux/mdate-sh "$self_abspathname" | sed -e 's,^.* ,,'`
   echo "\
-gnulib-tool (GNU $package $date)$version
+gnulib-tool (GNU $package $date)$version Debian: $debversion
 Copyright (C) $year Free Software Foundation, Inc.
 License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>
 This is free software: you are free to change and redistribute it.
